### ISSUE TITLE
Incompatible Types using Storage

### ISSUE THREAD

**Issue by cschroeter** [nbubna/store#82]
```typescript
export function area(id: string, area: Storage): StoreAPI;
```
should use `Web Storage API interface` instead of 
```typescript
export interface Storage {
    length: number;
    clear(): void;
    getItem(key: string): string;
    key(index: number): string;
    removeItem(key: string): void;
    setItem(key: string, value: string): void;
}
```

So 
```typescript
store.area("local", window.localStorage)
```
does not  have this TypeScript error

```typescript
The types returned by 'getItem(...)' are incompatible between these types.
    Type 'string | null' is not assignable to type 'string'.
      Type 'null' is not assignable to type 'string'.ts(2345)
```

**Comment by nbubna** [comment]
Yeah, you're right. I'm crazy busy right now. Are you up for creating a PR or patch to make this quick?

**Comment by cschroeter** [comment]
@nbubna is there a chance we can craft a release with the new typings? :P

**Comment by nbubna** [comment]
Yes, sorry. Life a bit nuts, maybe tonight?

**Comment by cschroeter** [comment]
@nbubna if you tell me what to do I also can do a release :)

**Comment by nbubna** [comment]
I published 2.12.0 this morning. Sorry for the delay! Life's a bit nuts
'round here right now.

**Comment by cschroeter** [comment]
@nbubna thanks - I hope things work out for you.
